lib/active_shipping/package.rb

```diff
@@ -1,118 +1,62 @@
-      end
-
-      @weight_unit_system = weight_imperial ? :imperial : :metric
-      @dimensions_unit_system = dimensions_imperial ? :imperial : :metric
-
-      @weight = attribute_from_metric_or_imperial(grams_or_ounces, Measured::Weight, @weight_unit_system, :grams, :ounces)
+      @length = length
+      @width  = width
+      @height = height || ( cylinder? ? width : Measured::Length.new(0, length.unit) )
 
-      if @dimensions.blank?
-        zero_length = Measured::Length.new(0, (dimensions_imperial ? :inches : :centimetres))
-        @dimensions = [zero_length] * 3
-      else
-        process_dimensions
-      end
+      @value = self.class.cents_from(options[:value])
+      @currency = options[:currency] || options[:value].try(:currency)
+    end
 
-      @value = Package.cents_from(options[:value])
-      @currency = options[:currency] || (options[:value].currency if options[:value].respond_to?(:currency))
-      @cylinder = (options[:cylinder] || options[:tube]) ? true : false
-      @gift = options[:gift] ? true : false
-      @oversized = options[:oversized] ? true : false
-      @unpackaged = options[:unpackaged] ? true : false
+    def dimensions
+      [@length, @width, @height]
     end
 
-    def unpackaged?
-      @unpackaged
+    def girth
+      @girth ||= cylinder ? width.scale(Math::PI) : (width + height).scale(2)
     end
+    alias_method :circumference, :girth
+    alias_method :around, :girth
 
-    def oversized?
-      @oversized
+    def volume
+      @volume ||= if cylinder?
+        Math::PI * width.scale(0.5).value * height.scale(0.5).value * length.value
+      else
+        length.value * width.value * height.value
+      end
     end
 
     def cylinder?
-      @cylinder
+      @cylinder ||= @options[:cylinder].present?
     end
     alias_method :tube?, :cylinder?
 
     def gift?
-      @gift
+      @gift ||= @options[:gift].present?
     end
 
-    def ounces(options = {})
-      weight(options).convert_to(:oz).value
-    end
-    alias_method :oz, :ounces
-
-    def grams(options = {})
-      weight(options).convert_to(:g).value
-    end
-    alias_method :g, :grams
-
-    def pounds(options = {})
-      weight(options).convert_to(:lb).value
-    end
-    alias_method :lb, :pounds
-    alias_method :lbs, :pounds
-
-    def kilograms(options = {})
-      weight(options).convert_to(:kg).value
-    end
-    alias_method :kg, :kilograms
-    alias_method :kgs, :kilograms
-
-    def inches(measurement = nil)
-      @inches ||= @dimensions.map { |m| m.convert_to(:in).value }
-      measurement.nil? ? @inches : measure(measurement, @inches)
-    end
-    alias_method :in, :inches
-
-    def centimetres(measurement = nil)
-      @centimetres ||= @dimensions.map { |m| m.convert_to(:cm).value }
-      measurement.nil? ? @centimetres : measure(measurement, @centimetres)
+    def oversized?
+      @oversized ||= @options[:oversized].present?
```

Comment:
Again I think we should look at what this is describing. It's a carrier's view of physical rating characteristics of a package.

e.g. for Canada Post it might be
```
options = {
  rating_characteristics: [:oversized]
}
```

Where as for USPS it might be

```
options = {
  rating_characteristics: [:oversized, :cubic]
```

Thoughts?